### PR TITLE
Meta: Safer text node targeting

### PR DESCRIPTION
**Pull request by fregante**
~~Untested.~~

This should avoid future issues where the text nodes change position and we silently remove the wrong element.

### CODE REVIEW

File: source/helpers/dom-utils.ts [hunk 1]
```diff
@@ -73,8 +73,26 @@ export const unhighlightTab = (tabElement: Element): void => {
...
+};
+
+export const removeTextNode = (node: Text | ChildNode, expectation: RegExp | string): void => {
+	const content = node.textContent!.trim();
+	if (!matchString(expectation, content)) {
+		throw new TypeError(`Expected node matching "${escapeMatcher(expectation)}", found "${content}"`);
+	}
+
+	node.remove();
 };
```
Comment (FloEdelmann) [inline source/helpers/dom-utils.ts:98]:
```suggestion
export const removeTextNode = (node: Text | ChildNode, expectation: RegExp | string): void => {
	assertNodeContent(node, expectation);
	node.remove();
};
```

Comment (fregante) [reply source/helpers/dom-utils.ts:98]: Oops, yes 😅

File: source/features/clean-repo-sidebar.tsx [hunk 2]
```diff
@@ -25,14 +25,6 @@ async function cleanReleases(): Promise<void> {
...
 		.firstElementChild! // About’s .BorderGrid-cell
 		.classList.add('border-0', 'pb-0');
-
-	// Align latest tag icon with the icons of other meta links
-	const tagIcon = select('.octicon-tag:not(.color-fg-success)', releasesSection)!;
-	if (tagIcon) {
-		tagIcon.classList.add('mr-2');
-		// Remove whitespace node
-		tagIcon.nextSibling!.remove();
-	}
```
Comment (fregante) [inline source/features/clean-repo-sidebar.tsx:35]: This doesn't seem to be hit anymore. The only tag icon I see is green.

Screen Shot


cc @cheap-glitch from https://github.com/refined-github/refined-github/pull/3974

File: source/features/useful-forks.tsx [hunk 3]
```diff
@@ -32,19 +32,20 @@ async function init(): Promise<void | false> {
...
 }
 
-function initArchivedRepoBanner(): void {
-	const banner = select('#js-repo-pjax-container > .flash-warn:first-child')!;
-	assertNodeContent(banner, 'repository has been archived');
-	attachElement(banner, {
-		append: createBannerLink,
-	});
+function initArchivedRepoBanner(signal: AbortSignal): void {
+	observe('[itemprop="name"] ~ .Label--attention', addLinkToBanner, {signal});
```
Comment (fregante) [inline source/features/useful-forks.tsx:48]: Found a way to make this feature instant

File: source/helpers/dom-utils.ts [hunk 4]
```diff
@@ -73,8 +73,29 @@ export const unhighlightTab = (tabElement: Element): void => {
...
+const matchString = (matcher: RegExp | string, string: string): boolean =>
+	typeof matcher === 'string' ? matcher === string : matcher.test(string);
+
+const escapeMatcher = (matcher: RegExp | string): string =>
+	typeof matcher === 'string' ? `"${matcher}"` : String(matcher);
+
+// eslint-disable-next-line @typescript-eslint/ban-types -- Nodes may be exactly `null`
+export const assertNodeContent = <N extends Text | ChildNode>(node: N | null, expectation: RegExp | string): N => {
+	if (!node || !(node instanceof Text)) {
+		console.warn('TypeError', node);
```
Comment (fregante) [inline source/helpers/dom-utils.ts:85]: I wish there was a way to attach data to errors and have the browser actually display it. `error.PROP` is basically inaccessible from the console, let alone _visible_

**Comment by fregante** [comment]
Everything tested. Merging 🚀

**Pull request merged**